geode-core/src/main/java/org/apache/geode/internal/cache/xmlcache/CacheXmlParser.java

```diff
@@ -642,6 +660,16 @@ private void startGatewaySender(Attributes atts) {
       gatewaySenderFactory.setPersistenceEnabled(Boolean.parseBoolean(enablePersistence));
     }
 
+    // Gateway-sender state
+    String state = atts.getValue(STATE);
+    if (Objects.equals(state, GatewaySenderState.RUNNING.getState()) ||
+        Objects.equals(state, GatewaySenderState.STOPPED.getState()) ||
+        Objects.equals(state, GatewaySenderState.PAUSED.getState())) {
+      gatewaySenderFactory.setState(GatewaySenderState.valueOf(state.toUpperCase()));
+    } else {
+      gatewaySenderFactory.setState(null);
```

Comment:
You probably should throw an exception here if an invalid state is used

Reply:
Yes, you are right! I will throw `InternalGemFireException`, since same is done for other parameters (e.g. orderPolicy).

Reply:
Yes, you are right. I will throw `InternalGemFireException`, since same is done for other parameters (e.g. orderPolicy).